internal/shim-sgx/src/sgx_uarch.rs

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! Contains microarchitecture structures of SGX. Keep the functionality
+//! minimal and stdlib free. Use the Intel SDM struct naming conventions,
+//! instead of Rust naming conventions.
+//!
+//! Chapter 34 of the Intel SDM contains detailed documentation for these
+//! structs.
+
+
+#[allow(non_camel_case_types)]
```

Comment:
This doesn't do what you intend it to do.

If you want this to apply to all types in this file, it needs to be `#![allow(non_camel_case_types)]`. As it is currently coded, this only applies to `REPORT_BODY`.